benchmarks/time/src/main/scala/strawman/collection/immutable/HashSetBenchmark.scala

```diff
@@ -15,55 +15,68 @@ import scala.Predef.intWrapper
 @Measurement(iterations = 12)
 @State(Scope.Benchmark)
 class HashSetBenchmark {
-
-  @Param(scala.Array(/*"0", */"1", "2", "3", "4", "7", "8", "15", "16", "17", "39", "282", "73121", "7312102"))
+  @Param(scala.Array(/*"0", */"1", "2", "3", "4", "7", "8", "15", "16", "17", "39", "282", "4096", "131070", "7312102"))
   var size: Int = _
 
   var xs: HashSet[Long] = _
-  var xss: scala.Array[HashSet[Long]] = _
+  var zs: HashSet[Long] = _
   var zipped: HashSet[(Long, Long)] = _
   var randomIndices: scala.Array[Int] = _
-  var randomIndices2: scala.Array[Int] = _
-  var randomXss: scala.Array[HashSet[Long]] = _
+  def fresh(n: Int) = HashSet((1 to n).map(_.toLong): _*)
 
   @Setup(Level.Trial)
-  def initData(): Unit = {
-    def freshCollection() = HashSet((1 to size).map(_.toLong): _*)
-    xs = freshCollection()
-    xss = scala.Array.fill(1000)(freshCollection())
-    zipped = xs.map(x => (x, x))
+  def initTrial(): Unit = {
     if (size > 0) {
       randomIndices = scala.Array.fill(1000)(scala.util.Random.nextInt(size))
-      randomIndices2 = scala.Array.fill(1000)(scala.util.Random.nextInt(size))
-      randomXss = scala.Array.fill(1000)(freshCollection().take(scala.util.Random.nextInt(size)))
     }
   }
 
+  @Setup(Level.Invocation)
+  def initInvocation(): Unit = {
+    xs = fresh(size)
+    zs = fresh((size / 1000) max 2)
+    zipped = xs.map(x => (x, x))
+  }
+
   @Benchmark
-  def incl(bh: Blackhole): Unit = {
-    var ys = HashSet.empty[Long]
+  @OperationsPerInvocation(100)
+  def create(bh: Blackhole): Unit = {
     var i = 0L
-    while (i < size) {
-      ys += i
+    while (i < 100) {
+      bh.consume(fresh(size))
       i += 1
     }
-    bh.consume(ys)
   }
 
   @Benchmark
-  def concat(bh: Blackhole): Unit = bh.consume(xs ++ xs)
-
-  @Benchmark
-  def tail(bh: Blackhole): Unit = bh.consume(xs.tail)
+  @OperationsPerInvocation(1000)
+  def expand_incl(bh: Blackhole): Unit = {
+    var ys = xs
+    var i = 0L
+    while (i < 1000) {
+      ys += -i
+      i += 1
+    }
+    bh.consume(ys)
+  }
 
   @Benchmark
-  def init(bh: Blackhole): Unit = bh.consume(xs.init)
+  @OperationsPerInvocation(100)
+  def expand_concat(bh: Blackhole): Unit = {
+    var ys = xs
+    var i = 0L
+    while (i < 100) {
+      ys = ys ++ zs.map(i * _)
+      i += 1
+    }
```

Comment:
Why do we need to make a loop rather than just calling `xs ++ zs`? Why do we need to `map` `zs` elements?

Reply:
All elements of `zs` will already be present in `xs` so for `Set` based collections the concat will not add any elements unless `zs` is transformed in some way. I now realize that a better way would be to instead make all elements of `zs` negative at initialization (so that `zs` will never contain any elements already contained in `xs`). 

The looping is strictly not necessary but I added it so that the cost of the benchmark becomes higher compared to the cost of running `initInvocation` before each invocation (where `xs` and `zs` are created) which should improve the quality of the result.